app/src/main/java/org/simple/clinic/home/patients/PatientsScreenController.kt

```diff
@@ -96,26 +103,49 @@ class PatientsScreenController @Inject constructor(
         .flatMap {
           val user = userSession.loggedInUser().map { (user) -> user!! }
 
+          val setVerificationStatusMessageVisible = { loggedInStatus: User.LoggedInStatus, ui: Ui ->
+            when (loggedInStatus) {
+              NOT_LOGGED_IN, OTP_REQUESTED, VERIFYING_OTP -> ui.showUserStatusAsPendingVerification()
+              LOGGED_IN -> ui.hideUserApprovalStatus()
+            }
+          }
+
           Observables.combineLatest(user, hasUserDismissedApprovedStatusPref.asObservable())
               .map { (user, userDismissedStatus) ->
                 when (user.status) {
                   WAITING_FOR_APPROVAL -> { ui: Ui -> ui.showUserStatusAsWaiting() }
-                  DISAPPROVED_FOR_SYNCING -> { ui: Ui -> ui.hideUserApprovalStatus() }
+                  DISAPPROVED_FOR_SYNCING -> { ui: Ui -> setVerificationStatusMessageVisible(user.loggedInStatus, ui) }
                   APPROVED_FOR_SYNCING -> {
                     val twentyFourHoursAgo = Instant.now().minus(24, ChronoUnit.HOURS)
                     val wasApprovedInLast24Hours = twentyFourHoursAgo < approvalStatusUpdatedAtPref.get()
 
                     if (userDismissedStatus.not() && wasApprovedInLast24Hours) {
                       { ui: Ui -> ui.showUserStatusAsApproved() }
                     } else {
-                      { ui: Ui -> ui.hideUserApprovalStatus() }
+                      { ui: Ui -> setVerificationStatusMessageVisible(user.loggedInStatus, ui) }
                     }
                   }
                 }
               }
         }
   }
 
+  private fun showUserVerificationAlert(events: Observable<UiEvent>): Observable<UiChange> {
+    return events.ofType<ScreenCreated>()
+        .flatMap { userSession.loggedInUser() }
+        .filter { it is Just<User> }
+        .map { (user) -> user!!.loggedInStatus }
+        .buffer(2, 1)
+        .filter { it.size == 2 }
+        .filter { it[0] == OTP_REQUESTED && it[1] == LOGGED_IN }
+        .map {
+          { ui: Ui ->
+            ui.showUserVerifiedAlert()
+            ui.hideUserApprovalStatus()
```

Comment:
Because that is triggered and checked only on screen creates and activity resumes, this gets triggered when the user logged in status changes on OTP validation.

Reply:
Then shouldn't we add that trigger along with scree creates and activity resumes? We've so far kept functions grouped by their category of ui changes. Here it feels that the job of showing and hiding approval status is split across multiple functions.

Reply:
I'll need to rename that function to something like `displayUserApprovalOrVerificationStatus` also. Do you have any ideas on better names?

Reply:
Changing it to `displayUserAccountStatusNotification`. Will also change a couple of the related methods on screen.